File: fc-chat/demo/part3/ch3/04-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/outbound/RejectResponse.java
package net.prostars.messagesystem.dto.websocket.outbound;

import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.constant.UserConnectionStatus;

public class RejectResponse extends BaseMessage {

  private final String username;
  private final UserConnectionStatus status;

  public RejectResponse(String username, UserConnectionStatus status) {
    super(MessageType.REJECT_RESPONSE);
    this.username = username;
    this.status = status;
  }

  public String getUsername() {
    return username;
  }

  public UserConnectionStatus getStatus() {
    return status;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/outbound/RejectResponse.kt
package com.yscorp.withpush.messagesystem.dto.websocket.outbound

import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.constant.UserConnectionStatus

class RejectResponse(val username: String, val status: UserConnectionStatus) : BaseMessage(MessageType.REJECT_RESPONSE) {
}
